libcnb/src/util.rs

```diff
@@ -0,0 +1,36 @@
+/// Removes [`std::io::Error`] values from a [`Result`] that have the
+/// [`std::io::ErrorKind::NotFound`] error kind by replacing them with the default value for `T`.
+pub(crate) fn default_on_not_found<T: Default>(
+    result: Result<T, std::io::Error>,
+) -> Result<T, std::io::Error> {
+    match result {
+        Err(io_error) => match io_error.kind() {
+            std::io::ErrorKind::NotFound => Ok(T::default()),
+            _ => Err(io_error),
+        },
+        other => other,
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::util::default_on_not_found;
+    use std::io::ErrorKind;
+
+    #[test]
+    fn test_with_not_found_kind() {
```

Comment:
Do the names of the tests need updating now that the function is called `default_on_not_found`?

Also, should there be a test for the `Ok` case? (up to you)

Reply:
Oh, the names are new. It's a test with a `NotFound` kind and the other is with "other". I can add something for `Ok`.